packages/opentelemetry-plugin-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// mongodb.Server type is deprecated so every use trigger a lint error
+/* tslint:disable:deprecation */
+
+import { BasePlugin, NoopLogger } from '@opentelemetry/core';
+import {
+  Span,
+  SpanKind,
+  Logger,
+  Tracer,
+  CanonicalCode,
+} from '@opentelemetry/types';
+import * as mongodb from 'mongodb';
+import * as shimmer from 'shimmer';
+
+type Func<T> = (...args: unknown[]) => T;
+type MongoDB = typeof mongodb;
+interface MongoInternalCommand {
+  findandmodify: boolean;
+  createIndexes: boolean;
+  count: boolean;
+  ismaster: boolean;
+}
+
+/** MongoDB instrumentation plugin for OpenTelemetry */
+export class MongoDBPlugin extends BasePlugin<MongoDB> {
+  private readonly SERVER_FNS = ['insert', 'update', 'remove'];
+  private readonly CURSOR_FNS_FIRST = ['_find', '_getmore'];
+  protected _logger!: Logger;
+  protected readonly _tracer!: Tracer;
```

Comment:
Technically `_trace` is set (outside ctor) when calling enable method. So readonly is not right. WDYT ?